File: src/main/java/no/nav/syfo/api/ressurser/azuread/HistoryController.java
package no.nav.syfo.api.ressurser.azuread;

import no.nav.security.oidc.api.ProtectedWithClaims;
import no.nav.syfo.consumer.pdl.PdlConsumer;
import no.nav.syfo.domain.Fodselsnummer;
import no.nav.syfo.api.domain.RSHistorikk;
import no.nav.syfo.domain.model.Mote;
import no.nav.syfo.service.*;
import no.nav.syfo.consumer.veiledertilgang.VeilederTilgangConsumer;
import org.springframework.web.bind.annotation.*;

import javax.inject.Inject;
import java.util.ArrayList;
import java.util.List;

import static no.nav.syfo.api.auth.OIDCIssuer.AZURE;
import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;

@RestController
@ProtectedWithClaims(issuer = AZURE)
@RequestMapping(value = "/api/internad/historikk")
public class HistoryController {

    private VeilederTilgangConsumer tilgangService;

    private MoteService moteService;

    private PdlConsumer pdlConsumer;

    private HistorikkService historikkService;

    @Inject
    public HistoryController(
            VeilederTilgangConsumer tilgangService,
            MoteService moteService,
            PdlConsumer pdlConsumer,
            HistorikkService historikkService
    ) {
        this.tilgangService = tilgangService;
        this.pdlConsumer = pdlConsumer;
        this.moteService = moteService;
        this.historikkService = historikkService;
    }

    @GetMapping(produces = APPLICATION_JSON_VALUE)
    public List<RSHistorikk> getHistory(
            @RequestParam(value = "fnr") String fnr
    ) {
        Fodselsnummer personFnr = new Fodselsnummer(fnr);

        tilgangService.throwExceptionIfVeilederWithoutAccess(personFnr);

        List<Mote> moter = moteService.findMoterByBrukerAktoerId(pdlConsumer.aktorId(personFnr).getValue());
        List<RSHistorikk> historikk = new ArrayList<>();
        historikk.addAll(historikkService.opprettetHistorikk(moter));
        historikk.addAll(historikkService.flereTidspunktHistorikk(moter));
        historikk.addAll(historikkService.avbruttHistorikk(moter));
        historikk.addAll(historikkService.bekreftetHistorikk(moter));
        return historikk;
    }
}

File: src/main/kotlin/no/nav/syfo/history/controller/HistoryController.kt
package no.nav.syfo.history.controller

import no.nav.security.oidc.api.ProtectedWithClaims
import no.nav.syfo.api.auth.OIDCIssuer.AZURE
import no.nav.syfo.consumer.pdl.PdlConsumer
import no.nav.syfo.consumer.veiledertilgang.VeilederTilgangConsumer
import no.nav.syfo.domain.Fodselsnummer
import no.nav.syfo.history.HistorikkService
import no.nav.syfo.service.MoteService
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import javax.inject.Inject

@RestController
@ProtectedWithClaims(issuer = AZURE)
@RequestMapping(value = ["/api/internad/historikk"])
class HistoryController @Inject constructor(
    private val tilgangService: VeilederTilgangConsumer,
    private val moteService: MoteService,
    private val pdlConsumer: PdlConsumer,
    private val historikkService: HistorikkService
) {
    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getHistory(
        @RequestParam(value = "fnr") fnr: String
    ): List<RSHistorikk> {
        val personFnr = Fodselsnummer(fnr)

        tilgangService.throwExceptionIfVeilederWithoutAccess(personFnr)

        val moter = moteService.findMoterByBrukerAktoerId(pdlConsumer.aktorId(personFnr).value)

        return historikkService.opprettetHistorikk(moter) +
            historikkService.flereTidspunktHistorikk(moter) +
            historikkService.avbruttHistorikk(moter) +
            historikkService.bekreftetHistorikk(moter)
    }
}
